### DISCUSSION TITLE
EUR as currency?

### DISCUSSION THREAD

**Discussion by lars776** [davetapley/helium-tax#61] [Q&A] [answered]
Hi!
Great work so far. 
Please could you provide a second currency option? i am missing the Euro currency in this tool.

**Comment by davetapley** [comment]
It's a sensible request. Unfortunately the big problem is finding an API to get prices from.

To make matters worse we'd need to query the rate for every reward transaction (assuming we wanted it as accurate as possible).

Some ballpark math:

- Had approx. 1000 non-US 'successful' downloads last month (based on Google Analytics)
- Approx. 5000 reward transactions per hotspot per year (based on quick test of a couple)
- Gives approx. 5 million requests per transaction (generously assuming only one hotspot per download)

I checked [a few](https://fixer.io/product) [APIs](https://exchangeratesapi.io/pricing/) and that would be well in to 'enterprise' volume, so I'm assuming $100s per month.

I'll tweet them and see what they can offer, but I'm afraid it's unlikely to happen unless someone wants to offer it for free (and I don't want to turn this in to a business since there's already https://www.minertax.com/).

**Reply by lars776** [reply]
Hi!
Thank you for your long answer. I understand the problem with the enterprise-API and I have another suggestion:
Is it possible to collect all data in USD like now and in a second step you ask a currency translator and write the EUR currency in the csv?

**Reply by davetapley** [reply]
> Is it possible to collect all data in USD

It is _possible_, but I don't think it would be a good solution.
e.g. [from here](https://github.com/fxcm/MarketData#tickdata) you can download CSVs for each historical week:
https://tickdata.fxcorporate.com/EURUSD/2019/1.csv.gz

Just one week is 113 MB, so that would be over 5 GB to load a whole year.

Of course, I could write a backend to do this, but that's basically what the other APIs are
(and as a side project I'm trying to minimize complexity here).

**Reply by davetapley** [reply]
... I just noticed those CSVs also offer daily versions, e.g:
https://candledata.fxcorporate.com/D1/EURUSD/2019.csv.gz

this is **much** smaller (1MB for year),
but I don't know if day rate granularity would be good enough?

**Comment by davetapley** [comment, marked as answer]
# Workaround

Assuming **day granularity is good enough** then you can use this workaround:

1. Download `.csv` as normal

2. Import in to [Google Sheets](http://sheets.google.com/)

3. Add new column to right of column B

4. Paste this formula in empty cell C2 and drag down to fill all rows:

    ```
    =INDEX(GOOGLEFINANCE("Currency:USDEUR","close", DATEVALUE(MID(A2,1,10)), 1), 2,2)*B2
    ```

    ⚠️ if you are in a **non-English** locale then replace `,` with `;`, see [here](https://github.com/davetapley/helium-tax/discussions/61#discussioncomment-2184359).

---

This uses the daily `close` price. You can see other options [here](https://support.google.com/docs/answer/3093281?hl=en) under "for historical data".

Obviously please check the values make sense, and:
https://github.com/davetapley/helium-tax/blob/92543e0be6f260286032fca78d13bf5e3ebcee71/index.html#L145-L151

**Reply by lars776** [reply]
Thank you very much for this workaround. This would be okay for me, but it don't works correct:

time | usd |   | hnt | price
-- | -- | -- | -- | --
2022-01-06T19:04:22+01:00 | 0.0502177381225 | #FEHLER! | 0.00131105 | 38.30345

I had an error in the formula, the B2 is red highlighted.

**Reply by davetapley** [reply]
@lars776 see example here:
https://docs.google.com/spreadsheets/d/1mahU7NnD9FFKNF73tCBZDPEmmxO_D7ZjiZly3sjdOoE/edit?usp=sharing

It appears like this for me:
<img width="675" alt="Screen Shot 2022-01-17 at 9 17 27 AM" src="https://user-images.githubusercontent.com/48232/149805041-fe1622bc-d9ca-49a8-8394-d601d418b1e4.png">

**Comment by erdomatic** [comment]
hi dave! thx for your great work! i have the same error like lars:
![Screenshot (1)](https://user-images.githubusercontent.com/98405739/150998526-8c2ac773-3ac4-4ef4-9a38-63bf2fa0e175.png)

**Reply by davetapley** [reply]
@erdomatic Interesting, I wonder if it's an issue with the `DATEVALUE` / `DATWERT` function not being able to parse [ISO 8601 calendar dates](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates) when in German.

Can you figure out how to get Sheets in to US English, and try again with `DATEVALUE` instead of `DATWERT`? 🙏🏻

**Reply by erdomatic** [reply]
ok, i will try

**Reply by erdomatic** [reply]
thats it dave! thank you! now it works ;)

**Reply by Felix-ius** [reply]
The German version of google sheets uses ; instead of , so you have to replace them in the funkction first. Furthermore you should try converting all your values from . to , | You can do that with the search and replace option.

**Reply by davetapley** [reply]
Amazing! Thanks for the feedback @Felix-ius, I've made https://github.com/davetapley/helium-tax/issues/68 and I'll try and get to it soon.

**Reply by davetapley** [reply]
@Felix-ius and to be sure for the formula it would need to be ⬇️ ?

```
=INDEX(GOOGLEFINANCE("Currency:USDEUR";"close"; DATEVALUE(MID(A2;1;10)); 1); 2;2)*B2
```

**Reply by Felix-ius** [reply]
@davetapley yes, the formula looks good to me :)

**Comment by sisqonrw** [comment]
Hi i have problems with the decimal places. i have done it like the description. i am in germany. here my sheet.

https://docs.google.com/spreadsheets/d/1lczXhAtkeEPNuJrPh-6VXkTE2J4_4VSLFsaEhjrvtJ4/edit?usp=sharing

usd is '0.056
i need it as 0,056
how can i convert it. Without " ' " and , instead of .

**Reply by Felix-ius** [reply]
@sisqonrw Du hast wahrscheinlich das "Suchen & Ersetzten" Tool falsch benutzt. Man kann das aber auch mit einer Formel lösen. Wenn du der Anleitung gefolgt bist und deine Euro-Spalte die C-Spalte ist, musst du =INDEX(GOOGLEFINANCE("Currency:USDEUR";"close"; DATWERT(TEIL(A2;1;10)); 1); 2;2)*(WECHSELN(B2;".";",")) als Formel eingeben. Mit der Formel WECHSELN(B2;".";",") ersetzt du quasi den Punkt mit dem Komma.

(Falls das nicht klappt, schreib einfach nochmal oder füge mich zu deinem Google Sheet hinzu. Wenn man den Link benutzt muss man dich erst anfragen)
